src/pages/Search/SearchChangeApproverPage.tsx

```diff
@@ -0,0 +1,252 @@
+            return;
+        }
+
+        if (onyxReports?.size === selectedReports.length) {
+            isLoadingOnyxReports.current = false;
+            return;
+        }
+
+        if (isLoadingOnyxReports.current) {
+            return;
+        }
+
+        isLoadingOnyxReports.current = true;
+        for (const selectedReport of selectedReports) {
+            if (onyxReports?.has(selectedReport.reportID)) {
+                continue;
+            }
+            openReport(selectedReport.reportID);
+        }
+    }, [hasLoadedApp, onyxReports, selectedReports]);
+
+    const selectedPolicies = useMemo(() => {
+        const policies = new Map<string, Policy>();
+        for (const selectedReport of selectedReports) {
+            const policy = allPolicies?.[`${ONYXKEYS.COLLECTION.POLICY}${selectedReport.policyID}`];
+            if (policy?.id) {
+                policies.set(policy.id, policy);
+            }
+        }
+        return Array.from(policies.values());
+    }, [selectedReports, allPolicies]);
+
+    const changeApprover = useCallback(() => {
+        if (!selectedApproverType) {
+            setHasError(true);
+            return;
+        }
+
+        if (selectedApproverType === APPROVER_TYPE.ADD_APPROVER) {
+            const policiesToUpgrade = selectedPolicies.filter((policy) => !isControlPolicy(policy));
+            if (policiesToUpgrade.length > 1) {
+                // Bulk upgrade is not supported, so show a general page to guide the user to upgrade manually
+                Navigation.navigate(ROUTES.WORKSPACE_UPGRADE.getRoute(undefined, undefined, ROUTES.CHANGE_APPROVER_SEARCH_RHP));
+                return;
+            }
+            if (policiesToUpgrade.length === 1) {
+                Navigation.navigate(
+                    ROUTES.WORKSPACE_UPGRADE.getRoute(policiesToUpgrade.at(0)?.id, CONST.UPGRADE_FEATURE_INTRO_MAPPING.multiApprovalLevels.alias, ROUTES.CHANGE_APPROVER_SEARCH_RHP),
+                );
+                return;
+            }
+
+            Navigation.navigate(ROUTES.CHANGE_APPROVER_ADD_APPROVER_SEARCH_RHP);
+            return;
+        }
+
+        isSavingRef.current = true;
+        for (const selectedReport of selectedReports) {
+            const policy = allPolicies?.[`${ONYXKEYS.COLLECTION.POLICY}${selectedReport.policyID}`];
+            const report = onyxReports?.get(selectedReport.reportID);
+            if (!policy || !report) {
+                continue;
+            }
+
+            if (report.managerID !== currentUserDetails.accountID) {
+                const hasViolations = hasViolationsReportUtils(report.reportID, transactionViolations, currentUserDetails.accountID, currentUserDetails.email ?? '');
+                assignReportToMe(report, currentUserDetails.accountID, currentUserDetails.email ?? '', policy, hasViolations, isASAPSubmitBetaEnabled);
+            }
+        }
+
+        clearSelectedTransactions();
```

Comment:
Are there transactions selected? I would think we have reports not transactions selected.